flang/test/Driver/code-gen-aarch64.f90

```diff
@@ -1,11 +1,9 @@
-! TODO Add `-target <arch>` once that's available
-
 ! RUN: rm -f %t.o
-! RUN: %flang_fc1 -emit-obj %s -o %t.o
-! RUN: llvm-objdump --disassemble-all %t.o | FileCheck %s
+! RUN: %flang_fc1 -emit-obj -triple aarch64-unknown-linux-gnu %s -o %t.o
+! RUN: llvm-objdump --triple aarch64-unknown-linux-gnu --disassemble-all %t.o | FileCheck %s
 ! RUN: rm -f %t.o
-! RUN: %flang -c %s -o %t.o
-! RUN: llvm-objdump --disassemble-all %t.o | FileCheck %s
+! RUN: %flang -c --target=aarch64-unknown-linux-gnu %s -o %t.o
+! RUN: llvm-objdump --triple aarch64-unknown-linux-gnu --disassemble-all %t.o | FileCheck %s
```

Comment:
This change is surprisingly spartan, but I see the test is an empty program.

Now that the target is being correctly handled, shouldn't this test include just a little more to guarantee that X86 code isn't still coming out? This comment applies to several tests for ARM and X86 below as well.

Reply:
> This change is surprisingly spartan, but I see the test is an empty program.

I think that it's good to keep it minimal. The generated object file is actually quite large (it contains the runtime libs). Would expanding the input source file bring any benefits?  

> Now that the target is being correctly handled, shouldn't this test include just a little more to guarantee that X86 code isn't still coming out? 

I hope that we don't have to. If the generated code is X86 instead of AArch64, then `llvm-objdump --triple aarch64-unknown-linux-gnu` will print gibberish:

```
0000000000000000 <_QQmain>:
       0: c0 03 5f                     	rolb	$95, (%rbx)
       3: d6                           	<unknown>
```
By checking for `ret` or `retq`, we can be fairly confident that everything works fine.

Let me expand this test with negative checks so that we can have more confidence that it works as expected.